FirebaseSessions/Sources/SessionStartEvent.swift

```diff
@@ -256,7 +256,7 @@ class SessionStartEvent: NSObject, GDTCOREventDataObject {
     -> firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype {
     var subtype: firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype
 
-    #if os(iOS) && !targetEnvironment(macCatalyst)
+    #if os(iOS) && !targetEnvironment(macCatalyst) && (swift(<5.9) || !os(xrOS))
```

Comment:
Moving the swift check to a previous line should get rid of the warning, but at the possible cost of duplicated code for both the >=5.9 and < 5.9.

Reply:
cc @andrewheard

Reply:
@paulb777 Themis and I pair programmed the approach you suggested and it resolves the warnings on Xcode 14.2 (and still passes on the iOS and visionOS simulators in Xcode 15 b3).

I think we should probably add some comments to explain what's going on but (assuming CI passes) I think this approach seems reasonable. WDYT?

Reply:
Thanks Andrew and Themis. I think this is good for now. It would be nice to avoid the duplicated code pattern and perhaps at least the big switch should be refactored into a function.  Otherwise I've been experimenting with Auth as well and don't see an alternative.

Reply:
Np! Do you have a suggested structure for refactoring the switch into a function? I can split it out but I can't think of a way to avoid duplicating it in the separate function as well.

Reply:
Oh, right. I see that now. Sorry about the goose chase.